src/utils.h

```diff
@@ -1,5 +1,5 @@
 #pragma once
-
+#include "hll_common.h"
```

Comment:
You shouldn't need this here, right? You have forward declaration of `hyperloglog_t` just below.

I'm kind of vigilant about this file since it is included from the generated code; if in the long term we'll decide to implement code generation using e.g. LLVM, it would be easier if it only contained absolute minimum of declarations required by the generated code. Also, helps compilation speed now.